receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -193,16 +195,23 @@ func (tr *transaction) Commit() error {
 		}
 
 		adjustStartTimestamp(tr.metricBuilder.startTime, metrics)
-	} else {
-		// AdjustMetrics - jobsMap has to be non-nil in this case.
-		// Note: metrics could be empty after adjustment, which needs to be checked before passing it on to ConsumeMetrics()
-		metrics, _ = NewMetricsAdjuster(tr.jobsMap.get(tr.job, tr.instance), tr.logger).AdjustMetrics(metrics)
 	}
 
 	numPoints := 0
+	var md pdata.Metrics
 	if len(metrics) > 0 {
-		md := opencensus.OCToMetrics(tr.node, tr.resource, metrics)
+		md = opencensus.OCToMetrics(tr.node, tr.resource, metrics)
+		fixStaleMetrics(&md)
 		numPoints = md.DataPointCount()
+	}
+
+	if !tr.useStartTimeMetric {
+		// AdjustMetricSlice - jobsMap has to be non-nil in this case.
```

Comment:
The comment here refers to the renamed method AdjustMetricsSlice, but the code below still uses AdjustMetrics. Should one of these be updated?

Reply:
Good catch, is a remnant of automated refactor that renamed the prior `AdjustMetrics` implementation to `AdjustMetricSlice`.  I have removed these comments as this one adds no value and the one below it is outdated and incorrect.